clusterloader2/pkg/measurement/util/wait_for_conditions.go

```diff
@@ -0,0 +1,146 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2"
+)
+
+// WaitForGenericK8sObjectsOptions is an options object used by WaitForGenericK8sObjectsNodes methods.
+type WaitForGenericK8sObjectsOptions struct {
+	// GroupVersionResource identification of the resource to fetch.
+	GroupVersionResource schema.GroupVersionResource
+	// Namespaces from which to pick the resources.
+	Namespaces NamespacesRange
+	// SuccessfulConditions, conditions to look for in the objects denoting good objects.
+	// Formatted as `ConditionType=ConditionStatus`, e.g. `Scheduled=true`.
+	SuccessfulConditions []string
+	// SuccessfulConditions, conditions to look for in the objects denoting failed objects.
+	// Formatted as `ConditionType=ConditionStatus`, e.g. `Scheduled=true`.
+	FailedConditions []string
+	// MinDesiredObjectCount minimum number of objects that should contain successful or failed
+	// condition.
+	MinDesiredObjectCount int
+	// MaxFailedObjectCount maximum number of objects that could contain failed condition.
+	MaxFailedObjectCount int
+	// CallerName name of the caller.
+	CallerName string
+	// WaitInterval interval to wait in between the refreshes of the loop.
+	WaitInterval time.Duration
+}
+
+// NamespacesRange represents namespace range which which will be queried.
+type NamespacesRange struct {
+	Prefix string
+	Min    int
+	Max    int
+}
+
+// StringPrefix returns prefix which should be included in all logs.
+func (o *WaitForGenericK8sObjectsOptions) StringPrefix() string {
+	return fmt.Sprintf("%s: objects: %q, namespaces: %q", o.CallerName, o.GroupVersionResource.String(), o.Namespaces.String())
+}
+
+// String returns printable representation of the namespaces range.
+func (nr *NamespacesRange) String() string {
+	return fmt.Sprintf("%s-(%d-%d)", nr.Prefix, nr.Min, nr.Max)
+}
+
+// getMap returns a map with namespaces which should be queried.
+func (nr *NamespacesRange) getMap() map[string]bool {
+	result := map[string]bool{}
+	for i := nr.Min; i <= nr.Max; i++ {
+		result[fmt.Sprintf("%s-%d", nr.Prefix, i)] = true
+	}
+	return result
+}
+
+// WaitForGenericK8sObjectsNodes waits till the desired number of nodes is ready.
+// If stopCh is closed before all nodes are ready, the error will be returned.
+func WaitForGenericK8sObjectsNodes(ctx context.Context, dynamicClient dynamic.Interface, options *WaitForGenericK8sObjectsOptions) error {
+	dos, err := NewDynamicObjectStore(ctx, dynamicClient, options.GroupVersionResource, options.Namespaces.getMap())
```

Comment:
Nitpick: s/dos/store - sounds more explicit TBH (and it doesn't matter whether it's dynamic or not - what is essential is that it's a "listable" store).